### ISSUE TITLE
region_plot raises weird IndexError when y_min > y_max

### ISSUE THREAD

**Issue by seblabbe** [sagemath/sage#35032]
### Is there an existing issue for this?

- [X] I have searched the existing issues for a bug report that matches the one I want to file, without success.

### Did you read the documentation and troubleshoot guide?

- [X] I have read the documentation and troubleshoot guide

### Environment

```markdown
- **OS**: Ubuntu 20.04
- **Sage Version**: SageMath version 9.8.beta1, Release Date: 2022-09-29
```

### Steps To Reproduce

```
sage: region_plot(x+y>0, (x,-20,20), (y,20,-20))
```

### Expected Behavior

A `ValueError` saying that `y_min` and  `y_max` should satisfy `y_min < y_max`.

### Actual Behavior

```
IndexError                                
Traceback (most recent call last)
Input In [168], in <cell line: 1>()
----> 1 region_plot(x+y>Integer(0), (x,-Integer(20),Integer(20)), (y, Integer(20),-Integer(20)))

File ~/GitBox/sage/src/sage/misc/decorators.py:491, in options.__call__.<locals>.wrapper(*args, **kwds)
    489     options['__original_opts'] = kwds
    490 options.update(kwds)
--> 491 return func(*args, **options)

File ~/GitBox/sage/src/sage/plot/contour_plot.py:1727, in region_plot(f, xrange, yrange, plot_points, incol, outcol, bordercol, borderstyle, borderwidth, alpha, **options)
   1723 g._set_extra_kwds(Graphics._extract_kwds_for_show(options,
   1724                                                   ignore=['xmin', 'xmax']))
   1726 if neqs == 0:
-> 1727     g.add_primitive(ContourPlot(xy_data_array, xrange, yrange,
   1728                                 dict(contours=[-1e-20, 0, 1e-20],
   1729                                      cmap=cmap,
   1730                                      fill=True, **options)))
   1731 else:
   1732     mask = numpy.asarray([[elt > 0 for elt in rows]
   1733                           for rows in xy_data_array],
   1734                          dtype=bool)

File ~/GitBox/sage/src/sage/plot/contour_plot.py:82, in ContourPlot.__init__(self, xy_data_array, xrange, yrange, options)
     80 self.xy_data_array = xy_data_array
     81 self.xy_array_row = len(xy_data_array)
---> 82 self.xy_array_col = len(xy_data_array[0])
     83 GraphicPrimitive.__init__(self, options)

IndexError: index 0 is out of bounds for axis 0 with size 0
```

### Additional Information

The current error does not help the user to find the real problem.

**Comment by jnash10** [comment]
Hi @seblabbe,

I've modified the code to solve the issue in PR #35065.

This is the change in the 'setup_for_eval_on_grid' function in misc.py which Contour_plot uses to prepare for plotting and preprocesses the ranges including checking for errors like correct range format.

```
#to check if range is correctly defined by user. gives error is xmin>xmax or ymin>ymax
    if (ranges[0][-2]>ranges[0][-1]):
        raise ValueError("Range not correctly defined. xmin>xmax")

    if (ranges[1][-2]>ranges[1][-1]):
        raise ValueError("Range not correctly defined. ymin>ymax")
```

**Comment by jnash10** [comment]
The new input-output looks like this:
```
sage: x,y = var('x,y')
sage: region_plot(x+y>0, (-20,20), (20,-20))
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[2], line 1
----> 1 region_plot(x+y>Integer(0), (-Integer(20),Integer(20)), (Integer(20),-Integer(20)))

File ~/projects/sage/src/sage/misc/decorators.py:496, in options.__call__.<locals>.wrapper(*args, **kwds)
    494     options['__original_opts'] = kwds
    495 options.update(kwds)
--> 496 return func(*args, **options)

File ~/projects/sage/src/sage/plot/contour_plot.py:1688, in region_plot(f, xrange, yrange, plot_points, incol, outcol, bordercol, borderstyle, borderwidth, alpha, **options)
   1682 if not f:
   1683     return implicit_plot(feqs[0], xrange, yrange, plot_points=plot_points,
   1684                          fill=False, linewidth=borderwidth,
   1685                          linestyle=borderstyle, color=bordercol, **options)
-> 1688 f_all, ranges = setup_for_eval_on_grid(feqs + f,
   1689                                        [xrange, yrange],
   1690                                        plot_points)
   1691 xrange, yrange = [r[:2] for r in ranges]
   1693 xy_data_arrays = numpy.asarray([[[func(x, y)
   1694                                   for x in xsrange(*ranges[0],
   1695                                                    include_endpoint=True)]
   1696                                  for y in xsrange(*ranges[1],
   1697                                                   include_endpoint=True)]
   1698                                 for func in f_all[neqs::]], dtype=float)

File ~/projects/sage/src/sage/plot/misc.py:145, in setup_for_eval_on_grid(funcs, ranges, plot_points, return_vars, imaginary_tolerance)
    142     raise ValueError("Range not correctly defined. xmin>xmax")
    144 if (ranges[1][-2]>ranges[1][-1]):
--> 145     raise ValueError("Range not correctly defined. ymin>ymax")
    147 else:
    148     vars, free_vars = unify_arguments(funcs)

ValueError: Range not correctly defined. ymin>ymax

```

**Comment by jnash10** [comment]
Please check the latest PR #35077 as a fix to the issue.

This PR has a clean commit history.

Thank you, looking forward to contribute to SageMath regularly.

**Comment by jnash10** [comment]
Last PR was interfering with old if/else case.

Updated.

Also merged base/develop into the branch. So there are 2 commits in total. (fix + merge)
Hopefully that does not create an issue.

Hopefully everything is fine this time.

**Comment by kwankyu** [comment]
Fixed by #35113
